pkg/network/driver_interface.go

```diff
@@ -279,6 +339,43 @@ func resizeDriverBuffer(compareSize int, buffer []uint8) []uint8 {
 	return buffer
 }
 
+// GetDNS returns a raw IP packet that wraps a DNS packet
+func (di *DriverInterface) ReadDNSPacket(visit func([]byte, time.Time) error) (didRead bool, err error) {
+	var bytesRead uint32
+	var key uint32 // returned by GetQueuedCompletionStatus, then ignored
+	var ol *windows.Overlapped
+	err = windows.GetQueuedCompletionStatus(di.dnsIOCP, &bytesRead, &key, &ol, 0) 
```

Comment:
Is this intentionally non-blocking? It seems like this would end up polling and chewing up a lot of CPU.

Reply:
There is a time.Sleep(5ms) every time we run out of packets to read. This is not the most efficient way to do it,  but keeps parity with the linux implementation.